apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -361,6 +377,30 @@ export class WorkshopForm extends React.Component {
     return this.state.course && Subjects[this.state.course];
   }
 
+  // For summer workshops for CSP and CSA, only workshop admins can change
+  // the “Is this a virtual workshop” field within one month of the workshop
+  // in order to prevent people from changing this last minute without
+  // talking to us or the Friday institute.
+  checkCannotChangeIfWorkshopVirtual(sessions, course) {
+    return (
+      (course === COURSE_CSP || course === COURSE_CSA) &&
+      !this.props.permission.has(WorkshopAdmin) &&
+      sessions.some(this.workshopStartsWithinMonth)
+    );
+  }
+
+  // Returns whether today is within a month before the workshop start.
+  workshopStartsWithinMonth(session) {
+    const today = new Date();
+    const workshopDate = new Date(session.date);
+    const monthBeforeWorkshopDate = new Date(
+      workshopDate.getFullYear(),
+      workshopDate.getMonth() - 1,
+      workshopDate.getDate()
+    );
+    return monthBeforeWorkshopDate <= today && today <= workshopDate;
+  }
```

Comment:
I originally had this as a function in `workshop_serializer.rb` to quickly determine the start date of a workshop so I wouldn't have to iterate through the sessions to check but I ran into 2 issues. 1) In `workshop.rb`, the `workshop_starting_date` is determined by which session is listed first not which comes first chronologically (which probably wouldn't be a big deal since most people would probably list the first session first, but just in case). 2) When a user is first creating a workshop there is no workshop entry yet to see if the sessions are within a month or not, so I would have to check the ones listed in this `workshop_form` anyways. So instead of having separate logic for editing an existing workshop vs. creating a new one, I figured it would be simpler to just use the same process for both here.